buf/registry/descriptor/v1/file_descriptor_proto_service.proto

```diff
@@ -0,0 +1,130 @@
+import "buf/validate/validate.proto";
+import "google/protobuf/compiler/plugin.proto";
+import "google/protobuf/descriptor.proto";
+import "google/protobuf/field_mask.proto";
+
+option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/descriptor/v1";
+
+// Provide compiled FileDescriptorProtos for Modules, Labels, or Commits.
+service FileDescriptorProtoService {
+  // Get compiled FileDescriptorProtos for the given Module, Label, or Commit.
+  rpc GetFileDescriptorProtos(GetFileDescriptorProtosRequest) returns (GetFileDescriptorProtosResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+}
+
+message GetFileDescriptorProtosRequest {
+  // The reference to get FileDescriptorProtos for.
+  //
+  // See the documentation on ResourceRef for resource resolution details.
+  //
+  // Once the resource is resolved, the following content is returned:
+  //   - If a Module is referenced, the FileDescriptorProtos for the Commit of the default Label is returned.
+  //   - If a Label is referenced, the FileDescriptorProtos for the Commit of this Label is returned.
+  //   - If a Commit is referenced, the FileDescriptorProtos for this Commit is returned.
+  buf.registry.module.v1.ResourceRef resource_ref = 1 [(buf.validate.field).required = true];
+  // Exclude imports from the returned FileDescriptorProtos.
+  //
+  // A file is an import if:
+  //
+  //   - It comes from a dependency of the Module specified via the ResourceRef used in the request.
+  //     For example, if the ResourceRef specifies "buf.build/foo/bar", which depends on
+  //     "buf.build/foo/baz", all files from "buf.build/foo/baz" will be imports.
+  //   - It is a Well-Known Type, and the Well-Known Type was not part of the Module specified
+  //     via the ResourceRef used in the request. Well-Known Types are special in the Protobuf ecosystem,
+  //     and can be imported without being contained in a specified module.
+  //   - It contains symbols that are referenced by symbols specified in the request. For example,
+  //     if the request specified symbol "foo.v1.Foo" contained within "foo.proto", , which has a message
+  //     field of type "baz.v1.Baz", which is contained within "baz.proto", the the file "baz.proto"
+  //     will be marked as an import, regardless of which Module it derived from. If the symbols field
+  //     is not specified on the request, this situation will never occur.
+  //
+  // Note that if imports are excluded, the returned FileDescriptorProtos may not (and likely will not)
+  // be self-contained. If imports are included, all symbols referenced within the FileDescriptorProtos
+  // are contained within the returned FileDescriptorProtos.
+  bool exclude_imports = 2;
+  // Exclude SourceCodeInfo from the returned FileDescriptorProtos.
+  //
+  // SourceCodeInfo is optional additional information that is not required to describe a set
+  // of Protobuf APIs.
+  bool exclude_source_code_info = 3;
```

Comment:
Open question whether this should be `include` or `exclude` - we decided for the HTTP GET endpoint to make this `include`.

Reply:
In my mind, the decision on whether to call this `include` or `exclude` depends on whatever you want to be returned by default. If this should be returned by default, then `exclude` is right (opt-out). If it shouldn't be returned by default, then `include` is right (opt-in). To decide what the default would be, I would think about the anticipated use cases and whether a majority of those use cases would need or not need this info. (similarly for the other fields)

Reply:
Writing tools against descriptors, I find myself constantly forgetting that the info is not returned by default and I need to set a flag somewhere in the chain of tools to ensure the info is available. I'd rather need to ask for it to be stripped as an optimization than have to remember to include it, personally.

This goes for the other opt-in/out questions below, as well.

Reply:
Including by default is more convenient for use cases like the CLI. Excluding by default is better for constrained environments, where things suddenly stop working because a giant module with many comments shows up.

Giant modules have other downsides. Including by default is the least surprising behavior in my book.